chainer/training/extensions/evaluator.py

```diff
@@ -0,0 +1,168 @@
+import copy
+
+import six
+
+from chainer.dataset import convert
+from chainer.dataset import iterator as iterator_module
+from chainer import link
+from chainer import reporter as reporter_module
+from chainer.training import extension
+from chainer import variable
+
+
+class Evaluator(extension.Extension):
+
+    """Trainer extension to evaluate models on a validation set.
+
+    This extension evaluates the current models by a given evaluation function.
+    It creates a :class:`~chainer.Reporter` object to store values observed in
+    the evaluation function on each iteration. The report for all iterations
+    are aggregated to :class:`~chainer.DictSummary`. The collected mean values
```

Comment:
I think some users want to aggregate reported values with some reduction functions other than mean function. I my self encountered such situation, in which I calculated F1 value in each iteration and report it. As F1 value of whole dataset does not equal to the mean of F1 values of each minibatch, I would like to aggregate these F1 values in some other way. Do you think we should like custom `eval_loop`?